cli/src/cli_util.rs

```diff
@@ -3099,8 +3099,13 @@ pub struct EarlyArgs {
     // Option<bool>.
     pub no_pager: Option<bool>,
     /// Additional configuration options (can be repeated)
-    //  TODO: Introduce a `--config` option with simpler syntax for simple
-    //  cases, designed so that `--config ui.color=auto` works
+    ///
+    /// The name should be specified as TOML dotted keys. The value should be
+    /// specified in either TOML syntax or as a string. If string value contains
+    /// TOML constructs (such as array notation), it must be quoted.
```

Comment:
I think it would be good explicitly say that TOML syntax is attempted first. That means it's safe for scripts to use this option with valid TOML and be sure that it won't be interpreted as a string.

Reply:
Does this work?
```
    /// The value should be specified as a TOML expression. If string value
    /// doesn't contain any TOML constructs (such as array notation), quotes can
    /// be omitted.
```

Reply:
Yes, that makes sense. Thanks.